src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/Protocol.java

```diff
@@ -198,4 +199,21 @@ private JSONArray dataEntry(Row dataRow, Schema schema) {
         }
         return entry;
     }
+
+    public static Throwable unwrapCause(Throwable t) {
+        Throwable result = t;
+        int layer = 0;
+        // limit the unwrapping up to 5 layers
+        while (result != null && layer < 5) {
+            if (result instanceof ElasticsearchException) {
+                return result;
+            }
+            if (result.getCause() == null) {
+                return result;
+            }
+            result = result.getCause();
+            layer++;
+        }
+        return result;
+    }
```

Comment:
remove this?

Reply:
Some exceptions that wrapped ES exceptions as the causes are thrown out. This method is to recognize them.